### PR TITLE
[Dynamic Shape] Provide CreateDynamicSymbolToStaticNumberPass

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
New features

### PR changes
Others

### Description

pcard-76996

[Dynamic Shape] Provide CreateDynamicSymbolToStaticNumberPass, decouple CINN frontend and backend process

In the execution chain of dynamic shapes, the serial execution chain of `pir + prim + ShapeDialect + compiler front-end and back-end` is too long, resulting in a large number of dependencies for debugging the entire chain. Two passes, `DynamicToStatic` and `StaticToDynamic`, are provided to decouple the verification process of the compiler front-end and back-end, enabling parallel verification. This PR provides the `DynamicToStatic` pass, where the front-end executes dynamic shapes and the back-end executes static shapes.

动态形状的执行链条上，`pir + 组合算子 + ShapeDialect + 编译器前后端` **串行执行**链条过长，导致整个链条的调试工作依赖项极多，提供 `DynamicToStatic` 和 `StaticToDynamic` 两个 pass，解耦编译器前后端的验证过程，使得工作可以并行开展验证。本 PR 提供的是 `DynamicToStatic` pass，前端执行动态 shape，后端执行静态 shape

### CODE REVIEW

File: paddle/fluid/pybind/pir.cc [hunk 1]
```diff
@@ -1562,6 +1563,10 @@ void AddCinnPass(std::shared_ptr<PassManager> &pass_manager,  // NOLINT
   pass_manager->AddPass(pir::CreateBuildCinnPass());
 
   pass_manager->AddPass(cinn::dialect::ir::CreateDivideGroupOpToFusionOpPass());
+  if (auto pass = cinn::dialect::ir::CreateConvertDynamicToStaticDimPass()) {
+    pass_manager->AddPass(std::move(pass.value()));
+    has_dynamic_shape = false;
```
Comment (tc20042008) [inline paddle/fluid/pybind/pir.cc]: force_static_shape = true

Comment (jiahy0825) [reply paddle/fluid/pybind/pir.cc]: Thanks, done~

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/convert_dynamic_to_static_dim_pass.cc [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+CreateConvertDynamicToStaticDimPass() {
+  if (!GetGlobalDynamicToStaticDimMap()->has_value()) return std::nullopt;
+  return std::make_unique<ConvertDynamicToStaticDimPass>();
+}
+
+}  // namespace ir
+}  // namespace dialect
+}  // namespace cinn
+
+// REGISTER_IR_PASS(cinn_group_lowering, DivideGroupOpToFusionOpPass);
```
Comment (tc20042008) [inline paddle/cinn/hlir/dialect/operator/transforms/group_merge/convert_dynamic_to_static_dim_pass.cc:262]: remove

File: test/ir/pir/cinn/symbolic/CMakeLists.txt [hunk 3]
```diff
@@ -46,4 +46,17 @@ if(WITH_GPU)
   set_tests_properties(test_sub_graph_for_backend PROPERTIES LABELS
                                                              "RUN_TYPE=CINN")
 
+  add_test(
+    NAME test_sub_graph_for_frontend
+    COMMAND
+      ${CMAKE_COMMAND} -E env
+      PYTHONPATH=${CMAKE_BINARY_DIR}:${CMAKE_BINARY_DIR}/python/:$ENV{PYTHONPATH}
+      FLAGS_cinn_convert_dynamic_to_static_dim=S0:64
```
Comment (tc20042008) [inline test/ir/pir/cinn/symbolic/CMakeLists.txt:54]: FLAGS_cinn_convert_dynamic_to_static_dim -> FLAGS_cinn_convert_dynamic_dim_to_static

**Review by tc20042008 — APPROVED** [review]

File: paddle/cinn/hlir/dialect/operator/transforms/group_merge/convert_dynamic_to_static_dim_pass.cc (hunk 2 above)
Comment (jiahy0825) [reply paddle/cinn/hlir/dialect/operator/transforms/group_merge/convert_dynamic_to_static_dim_pass.cc:262]: Next PR fix this

File: test/ir/pir/cinn/symbolic/CMakeLists.txt (hunk 3 above)
Comment (jiahy0825) [reply test/ir/pir/cinn/symbolic/CMakeLists.txt:54]: Considering others can use D2S right now, maybe we can use `FLAGS_cinn_convert_dynamic_to_static_dim` temporarily.

**Pull request merged**